app/src/utils/data/contributions.ts

```diff
@@ -414,6 +416,38 @@ function filterGrantRoundsForContributions(
   return roundName;
 }
 
+export function filterMatchingPoolContributions(
+  grantRound: GrantRound,
+  roundDonations: Record<string, GrantRoundDonation[]>,
+  roundName: string | undefined,
+  roundLogoPtr: MetaPtr | undefined
+): ContributionsDetail[] {
+  if (!grantRound || !roundDonations) return [];
+  const roundAddress = grantRound.address;
+  const roundContributions = roundDonations[roundAddress];
+  if (!roundContributions || roundContributions?.length === 0) return [];
+  return roundContributions?.map((contribution: GrantRoundDonation) => {
+    const logo =
+      (roundLogoPtr && roundLogoPtr.protocol === 1 ? ptrToURI(roundLogoPtr) : false) || '/placeholder_grant.svg';
```

Comment:
`ptrToURI` returns null for unsupported protocols so the below should be easier to read and more scalable (i.e. automatically works when new protocol IDs are supported)

```suggestion
    const logo = roundLogoPtr && ptrToURI(roundLogoPtr) ? ptrToURI(roundLogoPtr) : '/placeholder_grant.svg';
```